packages/slider/index.tsx

```diff
@@ -0,0 +1,380 @@
+import {getCorrectEventName} from '@material/animation/util';
+import {EventType, SpecificEventListener} from '@material/base/types';
+import {MDCSliderAdapter, MDCSliderFoundation} from '@material/slider';
+import classnames from 'classnames';
+import React from 'react';
+
+const convertDashToCamelCase = (propName: string) =>
+  propName.replace(/-(\w)/g, (_, v) => v.toUpperCase());
+
+type MouseLikeEvent = React.MouseEvent | React.PointerEvent | React.TouchEvent;
+
+export interface SliderProps extends React.HTMLProps<HTMLDivElement> {
```

Comment:
Is it more appropriate for the HTMLDivElement to be HTMLInputElement?

Reply:
Hmm.  That's a good point.  The type of `step`, `min`, `max`, and `value` would need to change to `string`, but it might be good from a uniformity standpoint.

I think I'd considered `HTMLInputElement`, but went with `HTMLDivElement` because it doesn't actually use an `<input>` element, so we could potentially end up passing `HTMLInputElement`-specific props to the `<div>` via `otherProps` unless they're explicitly removed in the `props` destructuring.